### ISSUE TITLE
vmagent labelmap: allow to match metric name

### ISSUE THREAD

**Issue by jelmd** [VictoriaMetrics/VictoriaMetrics#1998]
**Is your feature request related to a problem? Please describe.**
Right now renaming labels via `labelmap` from scraped metrics is very primitive because of the "all or nothing" approach, i.e. not taking any context into consideration. So it is not possible to restrict the the labelmap action, which is bad.

E.g. vmagent scrapes a metric like `node_gfx{device='nvidia0',...}` and another `node_net{device='net0',...)`. Here it is not possible to restrict renaming `device` to `nic` for the `node_net` metric, it would rename it in the `node_gfx`, too. 

**Describe the solution you'd like**
Because `target_label` is currently not used within the `labelmap` action, it could be easily extended to use it as a regex context tied to `__name__`. I.e. if `target_label` is not specified, `labelmap` behaves as usual. Otherwise (if specified), it gets checked against `__name__` value. If it does'nt match, it returns immediately w/o any change to any label, otherwise (on match) it behaves as usual.

E.g.:
```
...
"metric_relabel_configs": [{
        "action": "labelmap",
        "target_label": "node_net.*",
        "regex": "device",
        "replacement": "nic"
}]
...
```

In the same way `labeldrop`, `labelkeep` and `labelmap_all` could be enhanced.

**Comment by valyala** [comment]
Another approach is to add `if` option, which could accept arbitrary PromQL series filters. For example:
```yml
- action: labelmap
  if: '{__name__=~"node_net.*"}'
  regex: "device"
  replacement: "nic"
```

**Comment by valyala** [comment]
@jelmd , the commit 3f49bdaeffa00888e3233453d4bc00630aa5910d adds conditional relabeling to VictoriaMetrics components, which support relabeling ([vmagent](https://docs.victoriametrics.com/vmagent.html), [vmalert](https://docs.victoriametrics.com/vmalert.html) and [a single-node VictoriaMetrics](https://docs.victoriametrics.com/)). Conditional relabeling can be applied to any relabeling rule by adding `if` filter with arbitrary [time series selector](https://prometheus.io/docs/prometheus/latest/querying/basics/#time-series-selectors) as described in the comment above.

This functionality will be included in the next release. In the mean time it can be tested by building `vmagent` from the commit 3f49bdaeffa00888e3233453d4bc00630aa5910d according to [these instructions](https://docs.victoriametrics.com/vmagent.html#how-to-build-from-sources).

**Comment by valyala** [comment]
VictoriaMetrics components gained support for `if` filter in relabeling rules starting from [v1.74.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). See [these docs](https://docs.victoriametrics.com/vmagent.html#relabeling). Closing this feature request as done.

**Comment by valyala** [comment]
FYI, the `if` filter could improperly work in VictoriaMetrics because of the bug, which has been fixed in 1bd7637fe1ad361c6f63f711e08b081d979cb13d . See [the corresponding bugreport](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/4806). The bugfix will be included in the next release. In the mean time it is possible to make custom build for `vmagent` from the commit 1bd7637fe1ad361c6f63f711e08b081d979cb13d according to [these docs](https://docs.victoriametrics.com/vmagent.html#how-to-build-from-sources).

**Comment by valyala** [comment]
FYI, the `if` filter should work properly starting from [v1.93.0](https://docs.victoriametrics.com/CHANGELOG.html).
